airflow/models/dag.py

```diff
@@ -341,7 +341,10 @@ def __init__(
 
         self.user_defined_macros = user_defined_macros
         self.user_defined_filters = user_defined_filters
-        self.default_args = copy.deepcopy(default_args or {})
+        if not isinstance(default_args, Dict) and default_args is not None:
+            raise TypeError("default_args must be a dictionary")
+        else:
+            self.default_args = copy.deepcopy(default_args or {})
```

Comment:
This is the only place `default_args` must be a dict, but please use the built-in class instead of the typing alias.

```suggestion
        if default_args and not isinstance(default_args, dict):
            raise TypeError("default_args must be a dict")
        self.default_args = copy.deepcopy(default_args or {})
```